dts/x86/arduino_101.dts

```diff
@@ -0,0 +1,29 @@
+/dts-v1/;
+
+#include "intel_curie.dtsi"
+
+/ {
+	model = "Intel Curie Arduino 101 board";
+	compatible = "intel,quark";
+
+	aliases {
+		uart_0 = &uart0;
+		uart_1 = &uart1;
+	};
+
+	chosen {
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+		zephyr,console = &uart1;
+	};
+};
+
+&uart0 {
+	status = "ok";
+	current-speed = <115200>;
+};
+
```

Comment:
yes it should. To add hw-flowcontrol uart.yaml in has to be changed in dts/common/. I would add uart0 support in arduino.dts in next patch. These set of patch intention was to set the baudrate for uart1 from device tree instead of Kconfig.